task1/main.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+)
+
+type answer struct {
+	Host       string      `json:"host"`
+	UserAgent  string      `json:"user_agent"`
+	RequestUri string      `json:"request_uri"`
+	Header     http.Header `json:"headers"`
+}
+
+func HandleAnsw(w http.ResponseWriter, r *http.Request) {
+	ans := answer{
+		r.Host,
+		r.UserAgent(),
+		r.RequestURI,
+		r.Header,
+	}
+
+	buf, err := json.Marshal(ans)
+	if err != nil {
+		log.Println(err)
```

Comment:
In case of error you should return error message to the client, and set status code to an error.